components/needs_team/needs_team.jsx

```diff
@@ -127,6 +127,21 @@ export default class NeedsTeam extends React.Component {
         TeamStore.saveMyTeam(team);
         BrowserStore.setGlobalItem('team', team.id);
         TeamStore.emitChange();
+
+        let channel;
+        if (identifier) {
+            const channelName = identifier.toLowerCase();
+            channel = ChannelStore.getByName(channelName);
```

Comment:
Is there a way to use whatever is in redux instead on using the old stores?

Reply:
My bad, I thought it's the easy fix. I've updated the PR using the selectId from mattermost-redux.